### DISCUSSION TITLE
'Prior' object has no attribute '_type'

### DISCUSSION THREAD

**Discussion by MauricioGS99** [Joshuaalbert/jaxns#81] [Q&A]
Hi, 
I have re-written my code to the new version of JAXNS as follows:

from jaxns import Prior
def patient_1_priors() -> PriorModelGen:
    p=yield Prior(tfp.distributions.Bernoulli(probs=0.7),name='disease')
    return p

def patient_1(p):
    disease = p
    treatment = tfp.distributions.Bernoulli(probs=0.4,name='treatment').sample()
    list_variables=[disease,treatment]
    list_variables_no_treatment=[disease]
    
    q=np.random.randint(0,1)
    bloodpressure = tfp.distributions.Normal(non_linear_fn(jnp.array(random.sample(list_variables_no_treatment,np.random.randint(0,len(list_variables_no_treatment)))),q*treatment),1,name='bloodpressure').sample()
    list_variables.append(bloodpressure)
    if q==1:
        list_variables_no_treatment.append(bloodpressure)
     
    q=np.random.randint(0,1)
    weight = tfp.distributions.Normal(non_linear_fn(jnp.array(random.sample(list_variables_no_treatment,np.random.randint(0,len(list_variables_no_treatment)))),q*treatment), 1,name='weight').sample()
    list_variables.append(weight)
    if q==1:
        list_variables_no_treatment.append(weight)
    
    q=np.random.randint(0,1)
    heartattack=tfp.distributions.Normal(non_linear_fn(jnp.array(random.sample(list_variables_no_treatment,np.random.randint(0,len(list_variables_no_treatment)))),q*treatment),1,name='heartattack').sample()
    list_variables.append(heartattack)
    if q==1:
        list_variables_no_treatment.append(heartattack)
    
    variables=[]
    list_variables=[disease,treatment,bloodpressure,weight,heartattack]
    
    for i in range(5,math.floor(k/2)):
        q=np.random.randint(0,1)
        vi=tfp.distributions.Normal(non_linear_fn(jnp.array(random.sample(list_variables,np.random.randint(0,len(list_variables)))),q*treatment), 1,'variable'+str(i)).sample()
        variables.append(vi)
        list_variables.append(vi)
        if q==1:
            list_variables_no_treatment.append(vi)
    
    for i in range(math.floor(k/2)+1,k):
        q=np.random.randint(0,1)
        vi=tfp.distributions.Normal(non_linear_fn(jnp.array(random.sample(list_variables_no_treatment,np.random.randint(0,len(list_variables_no_treatment)))),0*treatment), 1,'variable'+str(i)).sample()
        variables.append(vi)
        list_variables.append(vi)
        list_variables_no_treatment.append(vi)
            
    q=np.random.randint(0,1)
    vi=tfp.distributions.Normal(non_linear_fn(jnp.array(random.sample(list_variables,np.random.randint(0,k))),q*treatment), 1,'variable'+str(k)).sample()
    variables.append(vi)
    list_variables.append(vi)
    if q==1:
        list_variables_no_treatment.append(vi)
    new_list = map(math.log10, list_variables)
    return new_list                       

but when I do:
model_patient_1 = Model(prior_model=patient_1_priors,log_likelihood=patient_1)

I receive:
'Prior' object has no attribute '_type'

Any ideas why?

Thanks

**Comment by Joshuaalbert** [comment]
Hi @MauricioGS99 try upgrading to 2.4.0. Sorry I missed this post.
